client/src/main/python/slipstream/wrappers/CloudWrapper.py

```diff
@@ -77,16 +77,13 @@ def start_node_instances(self):
                                           util.toTimeInIso8601(time.time()))
             return
 
-        cloud_service_name = self._get_cloud_service_name()
-        user_info = self._get_user_info(cloud_service_name)
-
-        provisioning_stop_time = self._get_provisioning_timeout_time(user_info)
-
-        self._start_nodes_and_clients(user_info, nodes_instances)
+        self._start_nodes_and_clients(
+            self._get_user_info(self._get_cloud_service_name()), nodes_instances)
 
         self.clean_local_cache()
+
         nodes_instances_list = self._get_node_instances_to_follow(nodes_instances.keys())
-        self._check_provisioning(nodes_instances_list, provisioning_stop_time)
+        self._check_provisioning(nodes_instances_list, self._get_provisioning_timeout_time())
```

Comment:
I finally implemented the queryable state timeout wall-clock time based on the locally set state start time. The state start time always gets re-set at the begging of the state (in `MachineExecutor`).  Please have a look at d1cb34e